vcd/structure.go

```diff
@@ -200,6 +203,42 @@ func contains(sliceToSearch []string, searched string) bool {
 	return found
 }
 
+// jsonToCompactString transforms an unmarshalled JSON in form of a map of string->any to a plain string without any spacing.
+func jsonToCompactString(inputJson map[string]interface{}) (string, error) {
+	rawJson, err := json.Marshal(inputJson)
+	if err != nil {
+		return "", err
+	}
+	compactedJson := new(bytes.Buffer)
+	err = json.Compact(compactedJson, rawJson)
+	if err != nil {
+		return "", err
+	}
+	return compactedJson.String(), nil
+}
+
+// areMarshaledJsonEqual compares that two marshaled JSON strings are equal or not. Returns an error if something
+// wrong happens when compacting both for comparison.
+func areMarshaledJsonEqual(json1, json2 []byte) (bool, error) {
+	if !json.Valid(json1) {
+		return false, fmt.Errorf("not a valid JSON: '%s'", json1)
+	}
+	if !json.Valid(json2) {
+		return false, fmt.Errorf("not a valid JSON: '%s'", json2)
+	}
```

Comment:
Would possibly be good to differentiate these errors somehow (even "not a valid JSON 1", "not a valid JSON 2") could work as it would help troubleshooting should such an error be ever hit

Reply:
True, changed.